src/Storage.Interface/Models/PartyTypesAllowed.cs

```diff
@@ -1,36 +1,76 @@
+using System.Collections.Generic;
 using Newtonsoft.Json;
 
 namespace Altinn.Platform.Storage.Interface.Models
 {
     /// <summary>
-    /// Represents an set of settings where application owner can define what types of parties
-    /// that are allowed to instantiate an application.
+    /// Represents a set of settings where application owner can define what types of parties
+    /// that are allowed to be owners of an instance in an application.
     /// </summary>
+    /// <remarks>If all values are set to false (the default), then all types are allowed as owners</remarks>
     [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
     public class PartyTypesAllowed
     {
         /// <summary>
-        /// Gets or sets a value indicating whether a bankruptcy estate is allowed to instantiate.
+        /// Gets or sets a value indicating whether a bankruptcy estate is allowed to be the owner of an instance.
         /// </summary>
         [JsonProperty(PropertyName = "bankruptcyEstate")]
         public bool BankruptcyEstate { get; set; }
 
         /// <summary>
-        /// Gets or sets a value indicating whether an organisation is allowed to instantiate.
+        /// Gets or sets a value indicating whether an organisation is allowed to be the owner of an instance.
         /// </summary>
         [JsonProperty(PropertyName = "organisation")]
         public bool Organisation { get; set; }
 
         /// <summary>
-        /// Gets or sets a value indicating whether a person is allowed to instantiate.
+        /// Gets or sets a value indicating whether a person is allowed to be the owner of an instance.
         /// </summary>
         [JsonProperty(PropertyName = "person")]
         public bool Person { get; set; }
 
         /// <summary>
-        /// Gets or sets a value indicating whether a sub unit is allowed to instantiate.
+        /// Gets or sets a value indicating whether a sub unit is allowed to be the owner of an instance.
         /// </summary>
         [JsonProperty(PropertyName = "subUnit")]
         public bool SubUnit { get; set; }
+
+        /// <summary>
+        /// Gets or sets the list of allowed instantiator types for an application.
+        /// </summary>
+        [JsonProperty(PropertyName = "allowedInstantiatorPartyTypes")]
+        public List<InstantiatorPartyType> AllowedInstantiatorPartyTypes { get; set; }
+
+        /// <summary>
+        /// Gets or sets the list of allowed instantiator types for an application when instantiation is done by copy.
+        /// </summary>
+        [JsonProperty(PropertyName = "allowedCopyInstantiatorPartyTypes")]
+        public List<InstantiatorPartyType> AllowedCopyInstantiatorPartyTypes { get; set; }
+    }
+
+    /// <summary>
+    /// Enum representing the different types of instantiators.
+    /// </summary>
+    public enum InstantiatorPartyType
```

Comment:
enums are written as numbers with System.Text.Json, so we need some annotations (or just not use enums)